GVFS/GVFS.FunctionalTests/Program.cs

```diff
@@ -85,6 +85,12 @@ public static void Main(string[] args)
                 excludeCategories.Add(Categories.MacOnly);
             }
 
+            if (runner.HasCustomArg("--extra-only"))
+            {
+                includeCategories.Add(Categories.ExtraCoverage);
+                excludeCategories.Remove(Categories.FullSuiteOnly);

```

Comment:
Do we need a new `ExtraCoverage` category?  

I think if you updated this code to:

```
if (runner.HasCustomArg("--extra-only"))
{                
    includeCategories.Add(Categories. FullSuiteOnly);
```

It would only run the tests marked as `FullSuiteOnly` (this is how the `--windows-only` flag works).

Unfortunately the behavior of include/exclude is a bit weird.  I believe as soon as you set an include category only tests from the included categories will be run.

Reply:
Yeah but `FullSuiteOnly` is also in the `excludeCategories` so we would need to keep the `Remove` here otherwise I'm not sure how it would handle the same category in both.

Reply:
> Yeah but FullSuiteOnly is also in the excludeCategories so we would need to keep the Remove here otherwise I'm not sure how it would handle the same category in both.

If you put this case here could you avoid it being in the excludeCategories?

```
            if (runner.HasCustomArg("--full-suite") || runner.HasCustomArg("--extra-only"))
            {
                if (runner.HasCustomArg("--extra-only"))
                {
                    includeCategories.Add(Categories. FullSuiteOnly);
                }
                else
                {
                    Console.WriteLine("Running the full suite of tests");
                }

                GVFSTestConfig.GitRepoTestsValidateWorkTree = DataSources.AllBools;

                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
                {
                    GVFSTestConfig.FileSystemRunners = FileSystemRunners.FileSystemRunner.AllWindowsRunners;
                }
                else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
                {
                    GVFSTestConfig.FileSystemRunners = FileSystemRunners.FileSystemRunner.AllMacRunners;
                }
            }
            else
            {
                GVFSTestConfig.GitRepoTestsValidateWorkTree =
                    new object[]
                    {
                        new object[] { true }
                    };

                excludeCategories.Add(Categories.FullSuiteOnly);
                GVFSTestConfig.FileSystemRunners = FileSystemRunners.FileSystemRunner.DefaultRunners;
            }
```

(Or alternatively give it its own `else if` block?)

Reply:
Also, FWIW now that these tests are run on their own I like the `ExtraCoverage` name more than `FullSuiteOnly`, so maybe the category to keep is the new one you've added?

Reply:
Do we want the `GVFSTestConfig.GitRepoTestsValidateWorkTree`  to be all bools and with all the runners?  I was thinking to add it in the else like so:
```
            else
            {
                GVFSTestConfig.GitRepoTestsValidateWorkTree =
                    new object[]
                    {
                        new object[] { true }
                    };
               
                if (runner.HasCustomArg("--extra-only"))
                {
                    includeCategories.Add(Categories. FullSuiteOnly);
                }
                else
                {
                    excludeCategories.Add(Categories.FullSuiteOnly);
                }

                GVFSTestConfig.FileSystemRunners = FileSystemRunners.FileSystemRunner.DefaultRunners;
            }
```

I think we just need to run the tests so I'm not sure how important it is to make sure we have both for validate the work tree and all the runners.  What do you think?

Reply:
> I think we just need to run the tests so I'm not sure how important it is to make sure we have both for validate the work tree and all the runners. What do you think?

Ahh good point.  For regular PRs I don't think we need all the runners and work tree validation scenarios.

I think we should still run those as part of the release process though (cc: @jrbriggs)

Reply:
There is still the CI build that will run the full functional tests so we still run everything with every merge into master.